app.js

```diff
@@ -0,0 +1,189 @@
+            searchBar.addEventListener('keyup', () => {
+                const searchTerm = searchBar.value.toLowerCase();
+                for (let i = 0; i <= 99; i++) {
+                    const item = document.getElementById(`one-token-item-${i}`)
+                    const itemName = document.getElementById(`name-value-${i}`)
+                    const itemSymbol = document.getElementById(`symbol-value-${i}`)
+                    const itemTextName = itemName.textContent.toLowerCase();
+                    const itemTextSymbol = itemSymbol.textContent.toLowerCase();
+                    if (itemTextName.includes(searchTerm) || itemTextSymbol.includes(searchTerm)) {
+                        item.style.display = 'block';
+                    } else {
+                        item.style.display = 'none';
+                    }
+                }
+            }
+            );
+        });
+
+        const tokenChart = new Chart("myChart", {
+            type: "line",
+            data: {
+                labels: ['', '', '', '', '', '', '', '', '', ''],
+                datasets: []
+            },
+            options: {
+                legend: { display: true }
+            }
+        });
+
+        const addData = (chart, newData) => {
+            chart.data.datasets.push(newData)
+            chart.update();
+        }
+
+        const removeData = (chart, index) => {
+            chart.data.datasets.splice(index, 1);
+            chart.update();
+        }
+
+        const annihilateData = chart => {
+            chart.data.datasets = [];
+            chart.update();
+        }
+
+        const switcherFunction = () => {
+            const maxSelections = 5;
+            const checkboxesNode = document.querySelectorAll('input[type="checkbox"]');
+            const checkboxes = Array.from(checkboxesNode);
+            const parsedTokens = JSON.parse(tokens['data'])['data']
+            const colorScheme = ['red', 'blue', 'green', 'purple', 'gray']
+            const allData = parsedTokens
+                .map(({ name, priceUsd }) => ({ name, priceUsd }))
+            checkboxes.forEach(checkbox => {
+                checkbox.addEventListener('change', () => {
+                    let selectedCount = 0;
+                    checkboxes.forEach(cb => {
+                        if (cb.checked) {
+                            const dataIndex = cb.value
+                            const usdArray = []
+                            for (let i = 1; i <= 10; i++) {
+                                usdArray.push(allData[dataIndex].priceUsd)
+                            }
+                            const newData = {
+                                data: usdArray,
+                                label: allData[dataIndex].name,
+                                borderColor: colorScheme[selectedCount],
+                                fill: false
+                            }
+                            addData(tokenChart, newData)
```

Comment:
was able to get to this funny situation somehow...
<img width="894" height="511" alt="Screenshot 2025-09-10 at 15 08 37" src="https://github.com/user-attachments/assets/552433a8-9541-412a-8e0e-8878463599bd" />